src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3181,14 +3190,14 @@ internal override void ReportStateMachineSuspensionPointInsertedRudeEdit(List<Ru
 
             // Handle addition of using keyword to using declaration.
             if (insertedSuspensionPoint.IsKind(SyntaxKind.VariableDeclarator) &&
-                match.ReverseMatches.TryGetValue(insertedSuspensionPoint.Parent.Parent, out var oldLocalDeclaration) &&
+                match.ReverseMatches.TryGetValue(insertedSuspensionPoint.Parent!.Parent!, out var oldLocalDeclaration) &&
                 !((LocalDeclarationStatementSyntax)oldLocalDeclaration).AwaitKeyword.IsKind(SyntaxKind.AwaitKeyword))
             {
-                var newLocalDeclaration = (LocalDeclarationStatementSyntax)insertedSuspensionPoint.Parent.Parent;
+                var newLocalDeclaration = (LocalDeclarationStatementSyntax)insertedSuspensionPoint!.Parent!.Parent!;
 
                 diagnostics.Add(new RudeEditDiagnostic(
                     RudeEditKind.Insert,
-                    newLocalDeclaration.AwaitKeyword.Span,
+                    newLocalDeclaration!.AwaitKeyword.Span,
```

Comment:
>! [](start = 39, length = 1)

Is `!` required here?

Reply:
Good catch.

---
In reply to: [362900968](https://github.com/dotnet/roslyn/pull/40699#discussion_r362900968) [](ancestors = 362900968)